dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -56,6 +85,224 @@
     private final PermissionAPI permissionAPI;
     private final WorkflowImportExportUtil workflowImportExportUtil;
 
+    /**
+     * Finds the bulk actions based on the {@link BulkActionForm}
+     * @param user {@link User}
+     * @param bulkActionForm {@link BulkActionForm}
+     * @return BulkActionView
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public BulkActionView findBulkActions(final User user,
+                                          final BulkActionForm bulkActionForm) throws DotSecurityException, DotDataException {
+
+        return (UtilMethods.isSet(bulkActionForm.getContentletIds()))?
+                    // 1) case with contentlet ids
+                    this.findBulkActionByContentlets(bulkActionForm.getContentletIds(), user):
+                    // 2) case when the user checks all (using a query)
+                    this.findBulkActionByQuery(bulkActionForm.getQuery(), user);
+    }
+
+
+    /**
+     * inds the bulk actions based on the luceneQuery
+     * @param luceneQuery
+     * @param user
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    private BulkActionView findBulkActionByQuery(final String luceneQuery,
+                                                 final User user) throws DotDataException, DotSecurityException {
+        final String cleanedUpQuery = QueryUtil.removeQueryPrefix(luceneQuery);
+        //We should only be considering Working content.
+        final String query = "{\n"
+                + "    \"query\" : { \n"
+                + "        \"query_string\" : {\n"
+                + "            \"query\" : \"" + cleanedUpQuery + "\"\n"
+                + "        } \n"
+                + "\n"
+                + "    },\n"
+                + "    \"aggs\" : {\n"
+                + "        \"tag\" : {\n"
+                + "            \"terms\" : {\n"
+                + "                \"field\" : \"wfstep\",\n"
+                + "                \"size\" : 1000  \n"
+                + "            }\n"
+                + "        }\n"
+                + "    },\n"
+                + "\t\"size\":0   \n"
+                + "}";
+        final ESSearchResults results = this.contentletAPI.esSearch(query, false, user, false);
+
+        Logger.debug(getClass(),"luceneQuery: "+ cleanedUpQuery);
```

Comment:
remember on debug you can use the supplier   Logger.debug(getClass(), ()->"luceneQuery: "+ cleanedUpQuery);